src/group_theory/perm/cycle_type.lean

```diff
@@ -148,9 +102,12 @@ begin
   { exact λ f hf, (mem_append.mp hf).elim (h₂l₁ f) (h₂l₂ f) },
   { refine pairwise_append.mpr ⟨h₃l₁, h₃l₂, λ f hf g hg a, by_contra (λ H, _)⟩,
     rw not_or_distrib at H,
-    exact ((congr_arg2 disjoint h₁l₁ h₁l₂).mpr h a).elim
-      (ne_of_eq_of_ne ((mem_list_cycles_iff h₂l₁ h₃l₁ (h₂l₁ f hf) H.1).1 hf 1).symm H.1)
-      (ne_of_eq_of_ne ((mem_list_cycles_iff h₂l₂ h₃l₂ (h₂l₂ g hg) H.2).1 hg 1).symm H.2) }
+    rw [←ne.def, ←ne.def, ←mem_support, ←mem_support] at H,
```

Comment:
Why did this proof become longer?

Reply:
I rewrote it so that it would be somewhat more readable.